src/main/groovy/life/qbic/datamodel/datasets/datastructure/files/DataFile.groovy

```diff
@@ -0,0 +1,79 @@
+package life.qbic.datamodel.datasets.datastructure.files
+
+import groovy.transform.CompileStatic
+import groovy.transform.ToString
+
+/**
+ * A DataFile object represents a file
+ * with a name, a path and a fileType
+ *
+ * @author: Sven Fillinger
+ */
+@ToString(includeNames = true)
+// includes property names
+@CompileStatic
+class DataFile {
+
+    /**
+     * The file name
+     */
+    final private String name
+
+    /**
+     * The relative file path
+     */
+    final private String relativePath
+
+    /**
+     * The file type
+     */
+    final private String fileType
+
+    protected DataFile() {}
```

Comment:
Should the default constructor be available? This will probably lead to failure since final values are not initialized in this case.
```suggestion
    private DataFile() {}
```

Reply:
It needs to be, otherwise you cannot extend the class